format/udf-spec.md

```diff
@@ -0,0 +1,285 @@
+| *required*  | `format-version`  | `int`                  | Metadata format version (must be `1`).                                                                          |
+| *required*  | `definitions`     | `list<overload>`       | List of function [overload](#overload) entities.                                                                |
+| *required*  | `definition-log`  | `list<definition-log>` | History of [definition snapshots](#definition-log).                                                             |
+| *required*  | `max-overload-id` | `long`                 | Highest `overload-id` currently assigned for this UDF. Used to allocate new overload identifiers monotonically. |
+| *optional*  | `location`        | `string`               | Storage location of metadata files.                                                                             |
+| *optional*  | `properties`      | `map`                  | A string to string map of properties.                                                                           |
+| *optional*  | `secure`          | `boolean`              | Whether it is a secure function. Default: `false`.                                                              |
+| *optional*  | `doc`             | `string`               | Documentation string.                                                                                           |
+
+Notes:
+1. When `secure` is `true`,
+    - Engines **SHOULD NOT** expose the function definition through any inspection (e.g., `SHOW FUNCTIONS`).
+    - Engines **SHOULD** ensure that execution does not leak sensitive information through any channels, such as error messages, logs, or query plans.
+   
+### Overload
+
+Function overloads allow multiple implementations of the same function name with different signatures. Each overload has
+the following fields:
+
+| Requirement | Field name                 | Type                                                   | Description                                                                                   |
+|-------------|----------------------------|--------------------------------------------------------|-----------------------------------------------------------------------------------------------|
+| *required*  | `overload-id`              | `long`                                                 | Monotonically increasing identifier of this function overload.                                |
+| *required*  | `parameters`               | `list<parameter>`                                      | Ordered list of [function parameters](#parameter). Invocation order **must** match this list. |
+| *required*  | `return-type`              | `string` (Iceberg data type)                           | Return type. Example: `"string"`, `"struct<...>"`.                                            |
```

Comment:
If this is an Iceberg data type, then it can't be a `string` because complex Iceberg types are represented as objects:

```json
{
  "type": "list",
  "element-id": 3,
  "element-required": true,
  "element": "string"
}
```

Iceberg does not parse these types other than to/from the JSON representation and that representation requires field IDs. If we want to omit the field IDs then that's fine but I think we need to be more specific. We don't want to need to parse complex types as strings -- we would need to define the string format and that would duplicate the JSON representation.

Reply:
Made the change accordingly. Added the requirement for returned values.